tests/integration/test-server.js

```diff
@@ -35,6 +35,21 @@ export function serveTest (handler) {
             curSubtest = null;
             subtestCnt++;
           }
+        },
+        async asyncTest (name, subtest) {
+          if (!filter(name))
+            return;
+          curSubtest = name;
+          let resolve, reject;
+          let promise = new Promise((res, rej) => {
+            resolve = res;
+            reject = rej;
+          });
+          subtest(resolve, reject);
+          evt.waitUntil(promise);
```

Comment:
It would be interesting to know why this is needed since strictly speaking the test async function itself is blocking the main event handler (due to the `await t.asyncTest(...)`), so this shouldn't actually be needed?

Reply:
It's not strictly speaking necessary, you're right. I at first misunderstood how async tests work, and had a hard time debugging why things didn't work, which is why I introduced this explicit `asyncTest` method.

I think it'd actually be nice to remove the async support from the normal `test`, to make it harder to misuse that one—but that'd of course require converting all existing tests.

Reply:
The API pattern we follow is exactly the same as Node.js's native test runner - https://nodejs.org/docs/latest/api/test.html#test-runner.